cpp/test/CMakeLists.txt

```diff
@@ -17,7 +17,7 @@
 
 function(ConfigureTest)
 
-  set(options OPTIONAL LIB)
+  set(options OPTIONAL LIB EXPLICIT_INSTANTIATE_ONLY)
```

Comment:
Is `LIB` needed anymore now that `RAFT_COMPILED` is `PUBLIC`? We can just link to `raft::compiled` by default and we'll get that macro defined. This is out of scope for this PR though.

Reply:
I think `LIB` is still needed and that its use is orthogonal to whether `RAFT_COMPILED` is `PUBLIC`.

My understanding is that `LIB` is used to optionally link to `raft::compiled`. This means that some tests (like `LINALG_TEST`) can be built without first building `libraft.so`. We specifically need the optional `LIB` flag now for `EXT_HEADERS_TEST_IMPLICIT`. 

In this PR, as well as before, `RAFT_COMPILED` would be defined automatically when linking `raft::compiled`.